packages/next-codemod/bin/upgrade.ts

```diff
@@ -611,3 +617,89 @@ function writeOverridesField(
     }
   }
 }
+
+function warnDependenciesOutOfRange(
+  appPackageJson: any,
+  versionMapping: Record<string, { version: string; required: boolean }>
+) {
+  const allDependenciesToInstall = {
+    ...appPackageJson.dependencies,
+    ...appPackageJson.devDependencies,
+  }
+
+  const dependenciesOutOfRange = new Map<
+    string,
+    {
+      [dependency: string]: {
+        currentVersion: string
+        expectedVersionRange: string
+      }
+    }
+  >()
+
+  for (const dependency of Object.keys(allDependenciesToInstall)) {
+    let pkgJson
+    try {
+      pkgJson = require(
+        require.resolve(`${dependency}/package.json`, { paths: [cwd] })
```

Comment:
Good call. Forgot about the fact this needs to consider cwd. I don't think it'll work for Yarn's PnP but we can leave this out for now.